server/src/modules/campaign-monitor/campaign-monitor.service.ts

```diff
@@ -1192,14 +1193,24 @@ export default class CampaignMonitorService {
 	 *
 	 * @returns void
 	 * This function will handle part one of the nightly sync. It will check if there are users
+	 * First we reset all FAILED statusses to SYNCED so we can try them again.
 	 * that have the "TO_SYNC" status and will handle this in batches. Once all records have
 	 * been handled, all items should have the "SYNCED" status and will be ready for the validation
 	 * part of the nightly sync.
 	 */
 	static async startSyncUsersToCmPart1(): Promise<void> {
-		return new Promise((resolve) => {
-			whilst(this.validateSyncUsersToCm, this.syncUsersToCm, () => resolve());
-		});
+		try {
+			await DataService.execute(RESET_FAILED_STATUS_TO_SYNCED);
+
+			return new Promise((resolve) => {
+				whilst(this.validateSyncUsersToCm, this.syncUsersToCm, () => resolve());
+			});
```

Comment:
The async lib supports a promise variation of each function when you do not pass a callback

```suggestion
			await whilst(this.validateSyncUsersToCm, this.syncUsersToCm);
```